### DISCUSSION TITLE
generation settings when running in extension host vs not

### DISCUSSION THREAD

**Discussion by chris-miner** [mike-lischke/vscode-antlr4#207] [Q&A] [answered]
I have a sample antlr project which specifies "none" as the generation setting mode.  When I open the project directly in vs code with version 2.3.1 of the extension no code is generated.  However, if I open the project in the extension host, code is generated as if I had the generation mode set to "internal".

While stepping through the code I couldn't see that the mode parameter was consulted before generation.  Specifically, SourceContext::generate seems to generate Java source unconditionally on activation/construction of the ExtensionHost.

Later, when you save a grammar file ExtensionHost::regenerateBackgroundData is ultimately called, and that exits immediately on mode == "none".

Why the unconditional generation on extension activation?  Why doesn't this happen when opening an antlr project directly (ie not via extension host launch)?

**Comment by mike-lischke** [comment, marked as answer]
Good find. I changed that part not so long ago, without considering that code generation might be disabled. This is fixed now.
